simulation/model.py

```diff
@@ -24,6 +24,7 @@ def __init__(
         self,
         start_datetime: datetime.datetime,
         step_interval: int,
+        end_datetime: datetime.datetime,
```

Comment:
I'm not sure about this. Right now in `main` what determines the end date of the simulation is `start_date + (num_steps * step_interval)`, where `start_date` and `step_interval` are properties of `DomesticHeatingABM` and `num_steps` is an argument to `DomesticHeatingABM.run`.

I think it's confusing to have that and `end_datetime`, because `end_datetime` suggests the simulation would stop on that date but in fact it's dependent on `num_steps` and `step_interval`. What happens if `run` runs the simulation past `end_datetime`?

Reply:
Nothing happens if the simulation runs past this date. It's just used in the derivation of `model.get_heat_pump_price_discount_schedule`, which must contain a key on or after the maximum `model.current_datetime` in its current setup. It isn't passed as a simulation parameter, so can't be misunderstood to affect the simulation in this way?